main.go

```diff
@@ -0,0 +1,132 @@
+package main
+
+import (
+	"context"
+	stdlog "log"
+	"net/http"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/metalmatze/signal/healthcheck"
+	"github.com/metalmatze/signal/internalserver"
+	rulesspec "github.com/observatorium/api/rules"
+	"github.com/oklog/run"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/collectors"
+	"github.com/thanos-io/thanos/pkg/objstore/client"
+
+	"github.com/observatorium/rules-objstore/pkg/config"
+	"github.com/observatorium/rules-objstore/pkg/server"
+)
+
+func main() {
+	cfg, err := config.ParseFlags()
+	if err != nil {
+		stdlog.Fatal(err)
+	}
+
+	logger := log.NewLogfmtLogger(log.NewSyncWriter(os.Stderr))
+	if cfg.LogFormat == "json" {
+		logger = log.NewJSONLogger(log.NewSyncWriter(os.Stderr))
+	}
+
+	logger = level.NewFilter(logger, cfg.LogLevel)
+	if cfg.Name != "" {
+		logger = log.With(logger, "name", cfg.Name)
+	}
+
+	logger = log.With(logger, "ts", log.DefaultTimestampUTC, "caller", log.DefaultCaller)
+	defer level.Info(logger).Log("msg", "exiting")
+
+	reg := prometheus.NewRegistry()
+	reg.MustRegister(
+		collectors.NewGoCollector(),
+		//nolint:exhaustivestruct
+		collectors.NewProcessCollector(collectors.ProcessCollectorOpts{}),
+	)
+
+	healthchecks := healthcheck.NewMetricsHandler(healthcheck.NewHandler(), reg)
+
+	if cfg.Server.HealthcheckURL != "" {
+		// Checks if server is up.
+		healthchecks.AddLivenessCheck("http",
+			healthcheck.HTTPCheck(
+				cfg.Server.HealthcheckURL,
+				http.MethodGet,
+				http.StatusNotFound,
+				time.Second,
+			),
+		)
+	}
+
+	bkt, err := client.NewBucket(logger, cfg.BucketConfig, reg, cfg.Name)
+	if err != nil {
+		stdlog.Fatalf("creating object store bucket client: %v", err)
```

Comment:
Ideally you have only fatals/panics in `main` and dedicate `run() error` function, but not a biggy